US-017

FacundoBocalandro opened the pull request.

FacundoBocalandro committed:
Added:
-Search book table. Depending on its condition of admin or student, it renders a table with different actions.
-Generic table. Also changed copies table to be a generic table.

FacundoBocalandro committed:
Integrated activate/deactivate book

GustavoCarvallo commented on src/common/GenericTable/GenericTable.css:
```diff
@@ -0,0 +1,29 @@
+    height: 35px;
+}
+
+.generic-table-container .generic-table-row .generic-table-column {
```
Here on the `generic-table-column` (and also on the `generic-table-row`), you should handle the text-overflow.

This could be done by adding:
```
white-space: nowrap;
overflow: hidden;
text-overflow: ellipsis;
```
on this selector.

**Tip:** On the "sprint_1" branch (backend) the onStart feature is already merged so you can pull that and test it with some real-like data.

GustavoCarvallo commented on src/Book/SearchBook/ActivateOrDeactivateButton.tsx:
```diff
@@ -0,0 +1,76 @@
+    return (
+        <>
+            {renderModal()}
+            <i className={active ? "far fa-check-circle search-book-green-icon" : "fas fa-ban search-book-red-icon"}
```
This should be:
- If the book is active --> Show the red fa-ban to deactivate it.
- If the book is deactivate --> Show the green fa-ban to activate it

GustavoCarvallo commented on src/Book/Book.tsx:
```diff
@@ -95,31 +89,31 @@ const Book = (props: Props) => {
+                return <>
+                    {selectedBook &&
+                    <BookDetails isOpen={true} onClose={() => setStatus(SEARCH)} selectedBook={selectedBook}/>}
+                </>
```
The DETAILS couldn't be inside the SEARCH? 
Something like this:

```
case SEARCH:
    return (
        <div>
            <SearchBook isAdmin={props.isAdmin} handleOpenCreation={handleOpenCreation}
                        openBookDetails={openBookDetails}/>
            {selectedBook &&
            <BookDetails isOpen={true} onClose={() => setStatus(SEARCH)} selectedBook={selectedBook}/>}
        </div>

    )
```

This way when you render the details modal the book table won't disappear.

GustavoCarvallo commented on src/Book/SearchBook/SearchBookTable.tsx:
```diff
@@ -0,0 +1,62 @@
+                component: props.isAdmin ?
+                    (row => (
+                        <div className={'admin-search-actions'}>
+                            <ActivateOrDeactivateButton defaultValue={row.active} id={row.id}/>
```
[AWAIT TO CONFIRMATION TO MODIFY]
This should follow the same logic as the details modal. 
It should be only 1 "ActivateDeactivateModal" (preferable inside the SEARCH section as commented before for the DETAILS) and then on the `SearchBookTable` you open that modal.

This change will avoid rendering 10 uneccessary modals on the SearchBookTable.

GustavoCarvallo replied:
Decided not to do it for this version. Closing revision.

GustavoCarvallo reviewed (changes requested):
Please review the comments.

FacundoBocalandro committed:
Fixed PR issues.
Missing to implement rendering just one modal for all activate/deactivate actions.

GustavoCarvallo committed:
Merge branch 'sprint_1' into feat/US-017

# Conflicts:
#	src/Book/CreateOrEditBook.tsx

GustavoCarvallo reviewed (approved):
LGTM.
Great job at creating the SearchBook and the GenericTable component! :ok_hand:
